### DISCUSSION TITLE
batch exporting of members/loads

### DISCUSSION THREAD

**Discussion by JvdPoel** [Dlubal-Software/Dlubal_CSharp_Client#36] [General]
Currently the API doesn't support batch exports of objects. So the grasshopper performance is way slower compared to rfem 5 com interface.

When can we expect batching of objects in the api?

**Comment by jarabroz** [comment]
Hi @JvdPoel,

We have implemented a get methods for a particular object. In the medium term, we do not plan to change it.

The comparison of COM and WebService is a bit misleading. They are two different technologies. With COM you directly access objects in the memory of the application.  In WebService, objects need to be serialized to XML, sent via TCP/IP to application, received, deserialize from XML to an object. Then some validations and checks are run.  If we make "batching" of objects, it cloud lead to some speed gain, but it still needs to make all the steps described.

**Reply by JvdPoel** [reply]
Could it be an option that the API handles the actions async and returns a succeed or failed whenever the ingesting fails?

Right now we have our plugin working in both RFEM5 and 6. We use the same exporting logic as we do in RFEM5 but because eacht object takes significantly longer to be processed the RFEM6 api seems to be slower.

I do know some steps how we could mitigate this issue on the grasshopper side but the performance gain wouldnt match asynchronously sending the data.

**Comment by ZTThomasOberthanner** [comment]
I had a quick look at the git repo and can still not find a solution to read all member data at once from a model file. Running a web request for each single member of a large model is not an option, this is way too slow. This topic is quite old, so I would like to know if your plans may have changed in the meantime.
To be honest, my workflow (and the software I distribute) heavenly depends on the COM interface. A transition to RFEM6 is not possible at the moment.

BTW: I am wondering if this is a general problem due to the change in architecture? I guess that SOAP is just slower in general due to the XML conversion and also somehow limited in request size? As XML adds a lot of tags the size of the response is way bigger than the original data...
